File: clas12/rtpc/hit.pl
use strict;
use warnings;

our %configuration;

sub define_hit
{

		# uploading the hit definition
		my %hit = init_hit();
		$hit{"name"}            = "rtpc";
		$hit{"description"}     = "RTPC hit definitions";
		$hit{"identifiers"}     = "rtpc";
		$hit{"signalThreshold"} = "0.0*MeV";
		$hit{"timeWindow"}      = "0.00001*ns";
		$hit{"prodThreshold"}   = "0.1*mm";
		$hit{"maxStep"}         = "1*cm";
		$hit{"delay"}           = "0*ns";
		$hit{"riseTime"}        = "1*ns";
		$hit{"fallTime"}        = "1*ns";
		$hit{"mvToMeV"}         = 100;
		$hit{"pedestal"}        = -20;
		print_hit(\%configuration, \%hit);

}

File: basic_examples/1_Simple_detector/ex1_2_add_digitization/hit.py
from gemc_api_hit import *

def define_hit(configuration):

	hit = MyHit(name="ctof", description="ctof hit definitions", identifiers="paddle", SignalThreshold="0.5*MeV",
		TimeWindow="0*ns", ProdThreshold="1.*mm", MaxStep="1*cm", delay="50*ns",
		riseTime="5*ns", fallTime="8*ns", mvToMeV=100, pedestal=-20)
	print_hit(configuration, hit)
